base/linalg/uniformscaling.jl

```diff
@@ -171,6 +172,14 @@ function isapprox(J1::UniformScaling{T}, J2::UniformScaling{S};
             rtol::Real=Base.rtoldefault(T,S), atol::Real=0, nans::Bool=false) where {T<:Number,S<:Number}
     isapprox(J1.λ, J2.λ, rtol=rtol, atol=atol, nans=nans)
 end
+function isapprox(J::UniformScaling,A::AbstractMatrix;
+                  rtol::Real=rtoldefault(promote_leaf_eltypes(A),eltype(J)),
+                  atol::Real=0, nans::Bool=false, norm::Function=vecnorm)
+    n = checksquare(A)
+    Jnorm = norm == vecnorm ? abs(J.λ)*sqrt(n) : norm(J)
```

Comment:
`norm(J)` isn't right, because `norm` might be some arbitrary other function that doesn't know about `UniformScaling`.   I think you have to do something like:
```jl
Jnorm = norm === vecnorm ? abs(J.λ)*sqrt(n) : norm === Base.norm ? abs(J.λ) : norm(diagm(fill(J.λ, n)))
```
or alternatively use my `uniformscaling_norm` suggestion.  Defining a function like `uniformscaling_norm`, i.e. relying on dispatch, makes it possible to implement support for more norms.   In practice, however, I doubt anyone will call this with a norm other than `vecnorm` or `Base.norm`, and in any case the `diagm` fallback is only worse by a constant factor, so I don't think it is worth the code complexity to try and do better.